keras_cv/layers/preprocessing/mix_up.py

```diff
@@ -0,0 +1,95 @@
+import tensorflow as tf
+import tensorflow.keras as keras
+import tensorflow.keras.layers as layers
+from tensorflow.python.platform import tf_logging as logging
+
+
+class MixUp(layers.Layer):
+    """MixUp implements the MixUp data augmentation technique.
+
+    Args:
+        rate: Float between 0 and 1.  The fraction of samples to augment.
+        alpha: Float between 0 and 1.  Inverse scale parameter for the gamma distribution.
+            This controls the shape of the distribution from which the smoothing values are 
+            sampled.  Defaults 0.2, which is a recommended value when training an imagenet1k
+            classification model.
+        label_smoothing: Float between 0 and 1.  The coefficient used in label smoothing.  Default 0.0.
+    References:
+        https://arxiv.org/abs/1710.09412.
```

Comment:
Use markdown link (since we will use the docstrings to generate rendered docs pages in the future)